From bd88050c39d784de4191d263bde05886e5ba6cee Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 10 Jun 2014 17:40:32 -0700
Subject: [PATCH 448/827] drm: xilinx: dp: Return -ETIMEDOUT after waiting 2ms for aux reply

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Return -ETIMEDOUT if there's no aux reply for 2ms.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 37c4ce103a048b238d04907168d61bac0f5cf984)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_dp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
index 8b832d4..639532f 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
@@ -346,13 +346,14 @@ static int xilinx_drm_dp_aux_cmd_submit(struct xilinx_drm_dp *dp, u32 cmd,
 			  (bytes - 1) << XILINX_DP_TX_AUX_COMMAND_BYTES_SHIFT);
 
 	/* Wait for reply to be delivered upto 2ms */
-	for (i = 0; i < 2; i++) {
+	for (i = 0; ; i++) {
 		reg = xilinx_drm_readl(iomem, XILINX_DP_TX_INTR_SIGNAL_STATE);
 
 		if (reg & XILINX_DP_TX_INTR_SIGNAL_STATE_REPLY)
 			break;
 
-		if (reg & XILINX_DP_TX_INTR_SIGNAL_STATE_REPLY_TIMEOUT)
+		if (reg & XILINX_DP_TX_INTR_SIGNAL_STATE_REPLY_TIMEOUT ||
+		    i == 2)
 			return -ETIMEDOUT;
 
 		udelay(1000);
-- 
2.9.3

